controllers/release/release_adapter.go

```diff
@@ -228,24 +229,31 @@ func (a *Adapter) EnsureSnapshotEnvironmentBindingIsCreated() (results.Operation
 // EnsureTargetContextIsSet is an operation that will ensure that the targetContext of the adapter is not nil. It will
 // set it to a context from the workspace in the ReleasePlan if one is defined or to its own context if not.
 func (a *Adapter) EnsureTargetContextIsSet() (results.OperationResult, error) {
-	if a.targetContext == nil {
+	if _, isKcp := logicalcluster.ClusterFromContext(a.context); !isKcp {
+		a.targetContext = a.context
+		return results.ContinueProcessing()
+	}
+
+	// If the target namespace/workspace is not set we should get the target from the ReleasePlan
+	if a.release.Status.Target == (kcp.NamespaceReference{}) {
 		releasePlan, err := a.getReleasePlan()
 		if err != nil {
```

Comment:
Wondering if we could check here for the workspace to exist as well and mark the release as invalid if not. This is KCP code already so... why not ;)